lntest/harness_node_manager.go

```diff
@@ -134,6 +134,9 @@ func (nm *nodeManager) shutdownNode(node *node.HarnessNode) error {
 func (nm *nodeManager) restartNode(ctxt context.Context,
 	hn *node.HarnessNode, callback func() error) error {
 
+	// Make sure to register the node after it has been restarted.
+	defer nm.registerNode(hn)
```

Comment:
Just out of curiosity: Why is this needed? It doesn't look like we're calling into `shutdownNode()` here, which is what actually removes the node from the list of active nodes.

Reply:
yeah seems unnecessary - let me remove it and run the CI in the other branch again.

Reply:
you are right - think it's a mistake when debugging flakes, now removed!